gpsd 3.20 breaks gpsd_client

with file: `gpsd_client/src/client.cpp`: I get compile errorss:
```
gpsd_client/src/client.cpp:88:11: error: no match for ‘operator!’ (operand type is ‘timespec_t’ {aka ‘timespec’})
   88 |       if (!p->online)
      |           ^~~~~~~~~~
gpsd_client/src/client.cpp:88:11: note: candidate: ‘operator!(bool)’ <built-in>
gpsd_client/src/client.cpp:88:11: note:   no known conversion for argument 1 from ‘timespec_t’ {aka ‘timespec’} to ‘bool’
gpsd_client/src/client.cpp: In member function ‘void GPSDClient::process_data_gps(gps_data_t*)’:
gpsd_client/src/client.cpp:156:27: error: cannot convert ‘timespec_t’ {aka ‘timespec’} to ‘gps_common::GPSFix_<std::allocator<void> >::_time_type’ {aka ‘double’} in assignment
  156 |         fix.time = p->fix.time;
      |                    ~~~~~~~^~~~
      |                           |
      |                           timespec_t {aka timespec}
gpsd_client/src/client.cpp:178:22: error: ‘struct gps_data_t’ has no member named ‘epe’
  178 |         fix.err = p->epe;
      |                      ^~~
gpsd_client/src/client.cpp: In member function ‘void GPSDClient::process_data_navsat(gps_data_t*)’:
gpsd_client/src/client.cpp:200:50: error: no matching function for call to ‘isnan(timespec_t&)’
  200 |       if (use_gps_time && !std::isnan(p->fix.time))
```
last version where it is buildable is 3.18 
but with 3.20 got errors related to changes in `gpsd` 
for example:
as mentioned here [https://fossies.org/linux/gpsd/gps.h](https://fossies.org/linux/gpsd/gps.h)
`gps_data_t` has changed and has no more a member `epe`

--- a/gpsd_client/src/client.cpp
+++ b/gpsd_client/src/client.cpp
@@ -85,8 +85,13 @@ class GPSDClient {
       if (p == NULL)
         return;
 
-      if (!p->online)
+#if GPSD_API_MAJOR_VERSION >= 9
+      if (!p->online.tv_sec && !p->online.tv_nsec) {
+#else
+      if (!p->online) {
+#endif
         return;
+      }
 
       process_data_gps(p);
       process_data_navsat(p);
@@ -153,7 +158,11 @@ class GPSDClient {
           status.status |= 18; // same here
 #endif
 
+#if GPSD_API_MAJOR_VERSION >= 9
+        fix.time = (double)(p->fix.time.tv_sec) + (double)(p->fix.time.tv_nsec) / 1000000.;
+#else
         fix.time = p->fix.time;
+#endif
         fix.latitude = p->fix.latitude;
         fix.longitude = p->fix.longitude;
         fix.altitude = p->fix.altitude;
@@ -188,7 +197,7 @@ class GPSDClient {
 
         /* TODO: attitude */
       } else {
-      	status.status = -1; // STATUS_NO_FIX
+        status.status = -1; // STATUS_NO_FIX
       }
 
       fix.status = status;
@@ -201,10 +210,17 @@ class GPSDClient {
 
       /* TODO: Support SBAS and other GBAS. */
 
-      if (use_gps_time && !std::isnan(p->fix.time))
+#if GPSD_API_MAJOR_VERSION >= 9
+      if (use_gps_time && (p->online.tv_sec || p->online.tv_nsec)) {
+        fix->header.stamp = ros::Time(p->fix.time.tv_sec, p->fix.time.tv_nsec);
+#else
+      if (use_gps_time && !std::isnan(p->fix.time)) {
         fix->header.stamp = ros::Time(p->fix.time);
-      else
+#endif
+      }
+      else {
         fix->header.stamp = ros::Time::now();
+      }
 
       fix->header.frame_id = frame_id;
 

